### PR TITLE
Add cast to xsd:dateTime

### LINKED ISSUE #190: date filters don't work on xsd:dateTime litterals
If you have a xsd:dateTime as range and you want to filter that with the sparnatural:TimeProperty-Date widget you will not get results.

The query contains `?dateTime1 <= "2021-01-19"^^xsd:date` which fails since the ?dateTime1 is of datatype xsd:dateTime.

This can be solved by wrapping the index in a casting function call, I'll submit a PR later this week.
the result will be `(xsd:date(?dateTime1)) <= "2021-01-19"^^xsd:date` which works as expected

### PR DESCRIPTION
**Pull request by semanticfire**
fixes #190
- Add proper cast on time filter

### CODE REVIEW

File: src/QueryGenerators.js [hunk 1]
```diff
@@ -308,8 +314,14 @@ class DefaultQueryGenerator {
...
-					""+index+"",
-					"\""+EndYear+"\"^^http://www.w3.org/2001/XMLSchema#date"
+					{
+						"type": "functioncall",
+						"function": "http://www.w3.org/2001/XMLSchema#dateTime",
+						"args" : [
+							""+index+""
+						]
+					},
+					"\""+EndYear+"\"^^http://www.w3.org/2001/XMLSchema#dateTime"
```
[the remark this answers is not in the document]
Comment (semanticfire) [reply src/QueryGenerators.js:324]: Works for Jena and Stardog, haven't tried anything else

[the remark this answers is not in the document]
Comment (semanticfire) [reply src/QueryGenerators.js:324]: Tested on Virtuoso works as well

[the remark this answers is not in the document]
Comment (tfrancart) [reply src/QueryGenerators.js:324]: Thanks ! let's merge

[the remark this answers is not in the document]
Comment (semanticfire) [reply src/QueryGenerators.js:324]: And GraphDB as well

**Pull request merged**